src/broker/module/broker_login_manager.cc

```diff
@@ -0,0 +1,18 @@
+#include "dsa_common.h"
+
+#include <dslink.h>
+#include "broker_login_manager.h"
+
+namespace dsa {
+
+void BrokerLoginManager::check_login(const string_ &username,
+                                     const string_ &password,
+                                     ClientInfo::GetClientCallback &&callback) {
+
```

Comment:
this should return the clientInfo too
also it's better to make the callback async